source/solvers/mf_navier_stokes.cc

```diff
@@ -1539,53 +1633,33 @@ template <int dim>
 void
 MFNavierStokesSolver<dim>::setup_GMG()
 {
-  gmg_preconditioner = std::make_shared<MFNavierStokesPreconditionGMG<dim>>();
+  this->computing_timer.enter_subsection("Setup GMG");
+
+  gmg_preconditioner = std::make_shared<MFNavierStokesPreconditionGMG<dim>>(
+    this->simulation_parameters,
+    this->dof_handler,
+    this->mapping,
+    this->cell_quadrature,
+    this->forcing_function,
+    this->simulation_control,
+    this->mg_computing_timer,
+    this->fe,
+    this->present_solution,
+    this->time_derivative_previous_solutions,
+    this->flow_control);
 
   if (this->simulation_parameters.linear_solver.at(PhysicsID::fluid_dynamics)
         .preconditioner == Parameters::LinearSolver::PreconditionerType::lsmg)
-    gmg_preconditioner->initialize_ls(this->computing_timer,
-                                      this->dof_handler,
-                                      this->simulation_parameters,
-                                      this->mapping,
-                                      this->fe,
-                                      this->mg_computing_timer,
-                                      this->cell_quadrature,
-                                      this->forcing_function,
-                                      this->present_solution,
-                                      this->time_derivative_previous_solutions,
-                                      this->pcout,
-                                      this->simulation_control,
-                                      this->flow_control);
+    gmg_preconditioner->initialize_ls(this->mg_computing_timer);
```

Comment:
I guess `initialize_ls()` and `initialize_gc()` can be merged, since `simulation_parameters` is passed to the constructor, which can be used to determine `lsmg` vs. `gcmg`.

Reply:
True